src/main/java/seedu/address/logic/commands/CommentCommand.java

```diff
@@ -0,0 +1,131 @@
+import seedu.address.model.person.PersonFactory;
+import seedu.address.model.person.Phone;
+
+/**
+ * Comments an existing person in the address book.
+ */
+public class CommentCommand extends Command implements ReversibleCommand {
+
+    public static final String COMMAND_WORD = "comment";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the comment of the person identified "
+            + "by the index number used in the displayed person list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) " + "COMMENT\n"
+            + "Example: " + COMMAND_WORD + " 1 " + "This is a comment.";
+
+    public static final String MESSAGE_COMMENT_PERSON_SUCCESS = "Commented Person: %1$s";
+    public static final String MESSAGE_SUCCESS_UNDO = "Changes reverted: %1$s";
+
+    private final Index index;
+    private final Comment editComment;
+
+    private Person originalPerson;
+    private Person editedPerson;
+
+    /**
+     * @param index of the person in the filtered person list to comment
+     * @param editComment comment to the person
+     */
+    public CommentCommand(Index index, Comment editComment) {
+        requireNonNull(index);
+        requireNonNull(editComment);
+        this.index = index;
+        this.editComment = editComment;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Person personToEdit = lastShownList.get(index.getZeroBased());
+        Person editedPerson = createEditedPerson(personToEdit, editComment);
+
+        model.setPerson(personToEdit, editedPerson);
+        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+
+        this.originalPerson = personToEdit;
+        this.editedPerson = editedPerson;
+
+        model.addCommand(this);
+        return new CommandResult(String.format(MESSAGE_COMMENT_PERSON_SUCCESS, Messages.format(editedPerson)));
+    }
+
+    @Override
+    public CommandResult undo(Model model) {
+        requireNonNull(model);
+
+        model.setPerson(editedPerson, originalPerson);
+        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+
+        return new CommandResult(String.format(CommentCommand.MESSAGE_SUCCESS_UNDO, Messages.format(originalPerson)));
+    }
+
+    @Override
+    public CommandResult redo(Model model) throws CommandException {
+        requireNonNull(model);
+
+        model.setPerson(originalPerson, editedPerson);
+        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+
+        return new CommandResult(String.format(MESSAGE_COMMENT_PERSON_SUCCESS, Messages.format(editedPerson)));
+    }
```

Comment:
These test cases are part of the main function and cannot be ignored. Please add them.